src/sagemaker/session.py

```diff
@@ -4630,6 +4654,35 @@ def delete_inference_component(self, inference_component_name: str, wait: bool =
                 poll=20,
             )
 
+    def list_and_paginate_inference_component_names_associated_with_endpoint(
```

Comment:
Same comment as above

Reply:
Session contains similar utilities for making SageMaker API calls. We want the code for making the client calls to be in the session since that'll have the proper credentials.